docs/docs/propeller/index.md

```diff
@@ -0,0 +1,36 @@
+---
+sidebar_position: 4
+---
+
+# Propeller
+
+Propeller is EquinoxJS's "[propulsion](https://github.com/jet/propulsion)
+lite". It's a set of abstractions for building reactive event processing
+pipelines catering specifically for event sourced reactions from Equinox's
+stores.
+
+## StreamsSink
+
+The `StreamsSink` receives batches of events from a source and intelligently
+manages handling them.
+
+- **Stream management**: maintains an ordered queue of incoming streams, ensuring
+  that events are processed in an orderly manner.
+- **Concurrency control**: To ensure optimal performance, StreamsSink restricts the
+  number of streams and event batches that can be processed at once. It
+  efficiently handles the inflow of new event batches, ensuring the system
+  doesn't get overwhelmed.
+- **Batch merging**: intelligently merges new events with existing streams. This
+  allows for a more streamlined event processing, ensuring that events from
+  multiple batches related to the same stream are handled together.
```

Comment:
>This
  allows for a more streamlined event processing, ensuring that events from
  multiple batches related to the same stream are handled together.
  
Where multiple batches include events from the same stream, these are supplied to the handler as a single sequence; this allows handler throughput to to be optimized (note the sink also guarantees that there will never be more than one handler processing events for a given stream)

(and if that last part is not true yet, it needs to be; probably worth making it a separate point so its easier to read and stands out)